cmake/FindLua.cmake

```diff
@@ -107,10 +107,14 @@ IF(LUA_LIBRARY)
 ENDIF(LUA_LIBRARY)
 
 # Determine Lua version
-IF(LUA_INCLUDE_DIR AND EXISTS "${LUA_INCLUDE_DIR}/lua.h")
-  FILE(STRINGS "${LUA_INCLUDE_DIR}/lua.h" lua_version_str REGEX "^#define[ \t]+LUA_RELEASE[ \t]+\"Lua .+\"")
-
-  STRING(REGEX REPLACE "^#define[ \t]+LUA_RELEASE[ \t]+\"Lua ([^\"]+)\".*" "\\1" LUA_VERSION_STRING "${lua_version_str}")
+IF(LUA_EXECUTABLE)
+  EXECUTE_PROCESS(COMMAND ${LUA_EXECUTABLE} -v
+                  OUTPUT_VARIABLE lua_version_str
+                  ERROR_VARIABLE lua_version_str)
```

Comment:
Just wanted to mention, for some reason Lua 5.1 seems to output the version string to `stderr`. At least that's the case for me under Win7. That's the reason for setting `ERROR_VARIABLE` to `lua_version_str` here, in case the interpreter splits out the version string to stderr, cmake will capture it.